server/src/transaction/transaction.service.ts

```diff
@@ -1,4 +1,107 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma.service';
+import { Prisma, Transaction, User } from '@prisma/client';
+import { Decimal } from '@prisma/client/runtime/library';
 
 @Injectable()
-export class TransactionService { }
+export class TransactionService { 
+  constructor(private prisma: PrismaService) {}
+
+  // Business logic methods
+
+  async getTransactionsForBalance(id: number): Promise<Transaction[] | null> {
+    return await this.prisma.transaction.findMany({
+      where: {OR: [
+        {account_id: id},
+        {transfer_id: id}
+      ]}
+    })
+  }
+
+  async sumOfTransactions(transactions: Transaction[], u_id: number): Promise<Prisma.Decimal | null> {
```

Comment:
to find the sum of transactions, we can use prisma's aggregations (to be implemented in transaction.prisma.service.ts). this is more efficient as well